### PR TITLE
PHP 8.3 support

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: .github/actions/test_tests-cache.sh [hunk 1]
```diff
@@ -3,7 +3,16 @@ set -e -u -x -o pipefail
 
 TMP_CACHE_DIR=$(mktemp -d -t glpi-cache-test-XXXXXXXXXX)
 
-for CONFIG in {"--use-default","--dsn=memcached://memcached","--dsn=redis://redis"}; do
+PHP_MAJOR_VERSION="$(echo $PHP_VERSION | cut -d '.' -f 1,2)"
+if [[ "$PHP_MAJOR_VERSION" -ne "8.3" ]]; then
+  CONFIGS=( "--use-default" "--dsn=memcached://memcached" "--dsn=redis://redis" )
+else
+  # test with redis adapter does not pass on PHP 8.3 alpha 1
+  # do not know if it is due to a PHP bug, a redis extension bug or a symfony bug
```
Comment (cedric-anne) [inline .github/actions/test_tests-cache.sh:11]: Maybe fixed by https://github.com/php/php-src/commit/49fbbea2ea5fe1f6bab0719c95a46a119ea3d91f

Waiting for next prerelease to be able to confirm.

Comment (cedric-anne) [reply .github/actions/test_tests-cache.sh]: Redis test suite is OK with PHP 8.3-alpha2.

**Comment by cedric-anne** [comment]
PHP 8.3 is now in beta version. No new change should be added now to PHP, except bug fixes.

**Review by orthagh — APPROVED** [review]

**Comment by cedric-anne** [comment]
I had to include the patch for `atoum` 4.1.0 -> 4.2.0 as the 4.2.0 has drop PHP 7.4 support (my fault).

**Review by trasher — APPROVED** [review]

**Pull request merged**